### DISCUSSION TITLE
Any known issues when loading a NativeAOT library in a .NET host?

### DISCUSSION THREAD

**Discussion by ulrichb** [dotnet/runtime#109416] [Q&A]
I did a few quick tests loading a (very simple) NativeAOT libarary (`[UnmanagedCallersOnly]`entrypoints) in a .NET (console) host program:

* Loading it in a .NET 8 (Windows, Linux and macOS) host and run some DLL entrypoints
  * Same with AOT compiled _host_
* Testing if debugging the host still works (only tested Windows)
* ...

Everything worked; still I'm wondering if there are any known issues.

The reason why I'm asking is the (old) thread #12018 about hosting multiple CLRs side-by-side. E.g. [this comment](https://github.com/dotnet/runtime/issues/12018#issuecomment-551214265) indicates issues with well-known names, "globally non-versioned named Win32 `HANDLE`s", etc.

Also I'm concerned about running the .NET GC multiple times in the sample process. Any possible conflicts there?

Bonus question: Are there any other known issues than the [documented](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/libraries) unsupported static libraries/library unloading and the general [AOT limitations](https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/?tabs=windows%2Cnet8#limitations-of-native-aot-deployment)? (I'm wondering if I missed some docs.)

**Comment by huoyaoyuan** [comment]
See discussions at #102048. NativeAOT compiled dll doesn't have most of CLR components.

**Reply by ulrichb** [reply]
Many thanks for the link ❤️. This covers part of my questions. 

Do you also have a reference for the "_NativeAOT compiled dll doesn't have most of CLR components_" part?

**Reply by huoyaoyuan** [reply]
It's more of a common sense about how nativeaot works. It doesn't contains nativeaot when referring to "CLR hosting". Nativeaot only uses the GC heap.
